generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -69,13 +70,14 @@ private boolean mustBeNull(FieldSpec fieldSpec) {
             && fieldSpec.getNullRestrictions().nullness == Nullness.MUST_BE_NULL;
     }
 
-    private List<FieldValueSource> getSetRestrictionSources(FieldSpec fieldSpec) {
-        List<Object> whitelist = new ArrayList<>(fieldSpec.getSetRestrictions().getWhitelist());
-         if (whitelist.isEmpty()){
-             return Collections.emptyList();
-         }
+    private List<FieldValueSource> getSetRestrictionSources(@Nullable Set<Object> whitelist) {
+        if (whitelist == null || whitelist.isEmpty()){
```

Comment:
can whitelist ever be null here?

Reply:
No - looks like I could flip that annotation.